### PR TITLE
[4.0] Fix installer file input width + other tweaks

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue #14657

### Summary of Changes

This PR:

- Fixes the installer file type input width issue so text doesn't overflow
- Replace a couple of instances of jQuery with vanilla JS
- Some styling tweaks
- Replace `<input type="button">` with a normal `<button>`
- Fixes the alert cog not directing the user to the options page.

### Testing Instructions

1. [**com_installer**] Ensure text doesn't overflow when using long file names
2. [**com_installer**] Go to the Folder tab, type in `http://google.com`, click `Install` and ensure the "loading" overlay still appears
3. [**com_installer**] Click the cog icon in the alert. This should now direct you to the correct page rather than just close the alert.
4. [**com_joomlaupdate**] Ensure text doesn't overflow when using long file names

### CODE REVIEW

**Comment by dgt41** [comment]
@C-Lodder can you create js files instead of the inline script?
Or in another PR

**Comment by C-Lodder** [comment]
@dgt41 - yeah sure

**Comment by franz-wohlkoenig** [comment]
@C-Lodder 1. + 2. works
3. you mean red Alert with "x" after 2.? 
4. you mean enter a long textstrin in "Install from Folder"?

**Review by dgrammatiko — APPROVED** [review]

**Comment by C-Lodder** [comment]
@frankmayer 

### 3:
screeny3


### 4:

No, attempt to install a zip file with a long name. The same styling is now being used on both `com_installer` and `com_joomlaupdate` so the result will be the same

**Comment by franz-wohlkoenig** [comment]
3. direct to `Upload Package File`, but Infortext is cutted:
bildschirmfoto 2017-03-16 um 17 26 38

4. works

If all is correct, i mark "successfully"

**Comment by C-Lodder** [comment]
That's a different issue. I believe it's BS related, not Joomla

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 89785b44cc712fd58d944989bfe499705a337a21<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14673">issues.joomla.org/tracker/joomla-cms/14673</a>.</sub>

**Comment by C-Lodder** [comment]
@dgt41 - moved com_installer JS and CSS to separate files

File: media/com_installer/js/installer.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			}
+		};
+
+	});
+
+})();
+
+
+jQuery(document).ready(function($) {
+
```
Comment (dgt41) [inline media/com_installer/js/installer.js]: No need for jQuery here, let me make a Pr against your branch

**Review by zero-24 — DISMISSED** [review]
Looks like there are missing ' ?

**Comment by C-Lodder** [comment]
@zero-24 - still tweaking this PR :)

**Comment by zero-24** [comment]
ok i remove my review than .

File: administrator/components/com_joomlaupdate/views/default/tmpl/default_upload.php [hunk 2]
```diff
@@ -22,7 +22,11 @@
 		}
 		else
 		{
-			jQuery("#loading").css("display", "block");
+			var loading = document.getElementById("loading");
+			if (loading)
+			{
+				loading.style.display = "block";
+			}
```
Comment (dgt41) [inline administrator/components/com_joomlaupdate/views/default/tmpl/default_upload.php]: If you add:
```php
JText::script('COM_INSTALLER_MSG_INSTALL_PLEASE_SELECT_A_PACKAGE', true);
```
then you can move this chunk of code to a js file
replacing
```js
alert("$errSelectPackage");
```
with
```js
alert(Joomla.JText._('COM_INSTALLER_MSG_INSTALL_PLEASE_SELECT_A_PACKAGE'));
```

File: administrator/components/com_installer/views/install/tmpl/default.php [hunk 3]
```diff
@@ -9,80 +9,18 @@
 
 defined('_JEXEC') or die;
 
+JHtml::_('behavior.core');
+JText::script('PLG_INSTALLER_PACKAGEINSTALLER_NO_PACKAGE');
```
Comment (dgt41) [inline administrator/components/com_installer/views/install/tmpl/default.php]: `JText::script('PLG_INSTALLER_PACKAGEINSTALLER_NO_PACKAGE', true);` makes sure that the string is going to be scaled correctly in js

**Comment by C-Lodder** [comment]
@zero-24 @franz-wohlkoenig all sorted now

**Comment by franz-wohlkoenig** [comment]
@C-Lodder test again?

**Comment by C-Lodder** [comment]
yes please

**Comment by franz-wohlkoenig** [comment]
fetching Data in Patchtester got at last 14699.

**Comment by C-Lodder** [comment]
conflicts fixed. There is an error installing an extension (`Call to undefined method JInstallerAdapterModule::get()`) however this isn't related to the PR

**Comment by franz-wohlkoenig** [comment]
fetching Data in Patchtester stopps at 14799.

**Comment by C-Lodder** [comment]
@franz-wohlkoenig How comfortable are you with terminal? You can simply fetch this PR using `git`

**Comment by franz-wohlkoenig** [comment]
uncomfortable. Independent my Comfort: If Testers should test, they need fetching all PR.

**Comment by mbabker** [comment]
Issues with patch tester should be reported on that extension's repo, they don't affect the validity of a PR here (especially as the patch tester isn't actually patching files, which creates a plethora of issues on its own).

**Comment by franz-wohlkoenig** [comment]
@mbabker right, i forgot the message about Patchtester in 4.0

**Comment by ciar4n** [comment]
I have tested this item :white_check_mark: successfully on ce2266ea8598811bc10ce58cc37bf7922abf36be<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/14673">issues.joomla.org/tracker/joomla-cms/14673</a>.</sub>

**Comment by C-Lodder** [comment]
@franz-wohlkoenig - would you mind retesting?

**Pull request merged**